sparsity/sparse_frame.py

```diff
@@ -8,12 +8,18 @@
 import pandas as pd
 from functools import partial, reduce
 from pandas.api import types
-from pandas.core.common import _default_index
 
 try:
-    from pandas.indexes.base import _ensure_index
+    # pandas>=0.24.0
+    from pandas.core.indexes.base import ensure_index
 except ImportError:
-    from pandas.core.indexes.base import _ensure_index
+    try:
+        # pandas>0.21.0
+        from pandas.core.indexes.base import _ensure_index
```

Comment:
Didn't you want to add `as ensure_index`? In other cases you import a name without underscore.

Reply:
It doesn't work even after fixing this:
```
TypeError: _convert_to_indexer() got an unexpected keyword argument 'raise_missing'
```

Reply:
Hm seems like this got introduced in a later stage...

Reply:
I removed it in a last commit to this branch (maybe not the best solution), but now tests fail. I'll leave the solution to you.